### PR TITLE
Index of person class is added

### PR DESCRIPTION
**Pull request by Reyano132**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-

## Checklist: I completed these to help reviewers :)

- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by HerbertYiga** [comment]
did you see the failing tavis?

File: api/src/main/java/org/openmrs/api/impl/PersonServiceImpl.java [hunk 1]
```diff
@@ -297,6 +297,7 @@ public void purgePerson(Person person) throws APIException {
 	public Person savePerson(Person person) throws APIException {
 		setPreferredPersonName(person);
 		setPreferredPersonAddress(person);
+		Context.updateSearchIndexForType(PersonAttribute.class);
```
Comment (kaweesi) [inline api/src/main/java/org/openmrs/api/impl/PersonServiceImpl.java]: are u here intending to update the index!! but i see your fields are on Person.class and no where on this class

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java [hunk 2]
```diff
@@ -747,6 +747,17 @@ public Long getCountOfPatients(String query, boolean includeVoided) {
 	}
 	
 	public List<Patient> findPatients(String query, boolean includeVoided, Integer start, Integer length){
+		//Just for testing...It has nothing to do with the function
+			List<String> testfields=new ArrayList<>();
+			testfields.add("gender");
+			LuceneQuery<Person> testQuery=LuceneQuery.newQuery(Person.class, sessionFactory.getCurrentSession(), "M", testfields);
```
Comment (kaweesi) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java]: can u try with already existing fields and see if this way of running the lucene query would work for any of them!

**Comment by dkayiwa** [comment]
@Reyano132 can you include the ticket id in the commit message as advised here? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Pull request closed without merging**